src/components/button/button_toggle/button_toggle.tsx

```diff
@@ -113,7 +112,7 @@ export const EuiButtonToggle: FunctionComponent<Props> = ({
         color={color}
         disabled={isDisabled}
         size={isIconOnly ? 's' : undefined} // only force small if it's the icon only version
-        {...rest}>
+        {...rest as Extract<EuiButtonTogglePropsForAnchor, EuiButtonTogglePropsForButtonToggle>}>
```

Comment:
TS's issue stems from it not understanding that `EuiButtonToggle`'s discriminator is the same as `EuiButton`'s, so it tries to match all the props button toggle _could_ have (e.g. `autoFocus`) to all variations in button.

`Extract<EuiButtonTogglePropsForAnchor, EuiButtonTogglePropsForButtonToggle>` returns only props that exist on both sides, satisfying `EuiButton`'s props.

TS is okay with this casting trick due to its usage of bivariance.